server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java

```diff
@@ -498,7 +498,10 @@ public class IoTDBConfig {
   private int maxInnerCompactionCandidateFileNum = 30;
 
   /** The max candidate file num in cross space compaction */
-  private int maxCrossCompactionCandidateFileNum = 1000;
+  private int maxCrossCompactionCandidateFileNum = 500;
```

Comment:
Suggest to change to `SingleCrossTaskCandidateFileLimit`

Reply:
Resolved.